examples/end2end_tree_reconstruction_test.py

```diff
@@ -0,0 +1,356 @@
+    except KeyError:
+        raise ValueError(f"Unsupported dataframe path extension: {ext}")
+
+
+def sample_reference_and_reconstruction(
+    differentia_bitwidth: int,
+    surface_size: int,
+    fossil_interval: typing.Optional[int],
+) -> ReconstructionResult:
+    """Sample a reference phylogeny and corresponding reconstruction."""
+    try:
+        paths = subprocess.run(
+            [
+                f"{os.path.dirname(__file__)}/"
+                "end2end_tree_reconstruction_with_dstream_surf.sh",
+                "--differentia-bitwidth",
+                f"{differentia_bitwidth}",
+                "--surface-size",
+                f"{surface_size}",
+                *(
+                    ["--fossil-interval", f"{fossil_interval}"]
+                    * (fossil_interval is not None)
+                ),
+            ],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout.strip()
+    except subprocess.CalledProcessError as e:
+        print(f"\033[33m{e.stdout}\033[0m")
+        print(f"\033[31m{e.stderr}\033[0m")
+        raise e
+
+    path_vars = dict()
+    exec(paths, path_vars)  # hack to load paths from shell script output
+    true_phylo_df = load_df(path_vars["true_phylo_df_path"])
+    reconst_phylo_df = alifestd_try_add_ancestor_list_col(
+        load_df(path_vars["reconst_phylo_df_path"]),
+    )  # ancestor_list column must be added to comply with alife standard
+
+    assert alifestd_count_leaf_nodes(
+        true_phylo_df
+    ) == alifestd_count_leaf_nodes(reconst_phylo_df)
+
+    reconst_phylo_df_extant = reconst_phylo_df.copy()
+    reconst_phylo_df_extant["extant"] = reconst_phylo_df["is_fossil"] == False
+    reconst_phylo_df_no_fossils = alifestd_prune_extinct_lineages_asexual(
+        reconst_phylo_df_extant
+    )
+
+    new_df = (
+        true_phylo_df.set_index("taxon_label")
+        .drop(
+            reconst_phylo_df["taxon_label"][reconst_phylo_df["is_fossil"] == True]  # type: ignore
+        )
+        .reset_index()
+    )
+
+    # true_phylo_df.to_csv("1.csv")
+    # new_df.to_csv("2.csv")
+    true_phylo_df_no_fossils = alifestd_prune_extinct_lineages_asexual(new_df)
+
+    return {
+        "true": true_phylo_df,
+        "reconst": reconst_phylo_df,
+        "true_dropped_fossils": true_phylo_df_no_fossils,
+        "reconst_dropped_fossils": reconst_phylo_df_no_fossils,
+    }
+
+
+def plot_colorclade_comparison(
+    frames: ReconstructionResult, *, fossils: bool
+) -> None:
+    plotter_kwargs = {
+        "taxon_name_key": "taxon_label",
+        "backend": "biopython",
+        "color_labels": "white",
+        "line_width": 2.5,
+    }
+    label_func = lambda node: (
+        " " + node.name if hash(node.name) % 12 == 0 else ""
+    )
+
+    plt.style.use("dark_background")
+    fig, axes = plt.subplots(3 if fossils else 2, 2)
+
+    frames["true"] = alifestd_collapse_unifurcations(frames["true"])
```

Comment:
ah, the "true" is a bit confusing to parse mentally due to also being a boolean; perhaps "actual" or "exact"?

Reply:
would have to change above, too